Parser/ABOParser.php

```diff
@@ -204,53 +179,223 @@ protected function parseTransactionLine($line)
         $transaction = $this->getTransactionClass();
 
         # Receipt ID
-        $receiptId = ltrim(substr($line, 35, 13), '0');
-        $transaction->setReceiptId($receiptId);
+        $transaction->setReceiptId($this->parseReceiptId($line));
 
         # Debit / Credit
-        $amount = ltrim(substr($line, 48, 12), '0') / 100;
-        $postingCode = substr($line, 60, 1);
-        switch ($postingCode) {
+        switch ($this->parsePostingCode($line)) {
             case self::POSTING_CODE_DEBIT:
-                $transaction->setDebit($amount);
+                $transaction->setDebit($this->parseAmount($line));
```

Comment:
Yeah. I didn't mean that because of performance but because of code. We can be at 100% sure that it's going to be called so call it before is not wrong, but in case of modification is easier to modify 1 line and I believe maybe even simpler to read variable then in all 4 cases thinking about what method is called and which arguments are passing. Current solution is not wrong at all, it's just suggestion how this can be "better" IMHO.

Btw.: there are more stuffs what I would done better in this library now, but until now I had no response that someone is using it :-)